lib/rules/jsx-filename-extension.js

```diff
@@ -32,9 +32,12 @@ module.exports = function(context) {
   return {
 
     JSXElement: function(node) {
-      var allowedExtensions = getExtensionsConfig();
       var filename = context.getFilename();
+      if (filename === '<text>') {
```

Comment:
I wonder if this comparison could reference a variable or something instead of hard-coding `<text>`